odoo/addons/base/models/res_partner.py

```diff
@@ -574,7 +574,8 @@ def create(self, vals_list):
     @api.ondelete(at_uninstall=False)
     def _unlink_except_internal_user(self):
         for partner in self:
-            if partner.active and partner.user_ids:
+            if self.env['res.partner'].sudo().search([('id', '=', partner.id)], limit=1, count=True) \
```

Comment:
to avoid making a search on id, you can directly use `browse()`.

Anyway, if you want to be sure the partner has already been created, you can see it in the id directly. (either it's a newId and it's not created yet in the DB, either it's a int and it already exists in the DB).
-> You can check this in testing `partner._origin.id`. If it's false,, the partner has not been created yet.

Is that what you meant ?

Reply:
I added this check to ensure that the partner in self is not already deleted (but the caller kept the reference to it), because if it is, the rest of the condition will throw an error as we reference nonexistent attributes (active, user_ids). 

`partner._origin.id`
`self.env['res.partner'].sudo().browse(partner.id)`

both throw an error in case partner was already deleted.

See this test file : odoo/odoo/addons/base/tests/test_orm.py :
```python
p1 = self.env['res.partner'].create({'name': 'W'})
[...]
p1.unlink()
[...]
# Deleting an already deleted record should be simply ignored
self.assertTrue(p1.unlink(), "Re-deleting should be a no-op")
```

Reply:
Try this:
<pre>
for partner in self.exists():
  if partner.active and partner.user_ids:
  .
  .
  .
</pre>